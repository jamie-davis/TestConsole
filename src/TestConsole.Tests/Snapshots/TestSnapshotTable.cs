using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using FluentAssertions;
using NUnit.Framework;
using TestConsoleLib;
using TestConsoleLib.Snapshots;
using TestConsoleLib.Testing;

namespace TestConsole.Tests.Snapshots
{
    [TestFixture]
    public class TestSnapshotTable
    {
        private SnapshotTable _table;

        [SetUp]
        public void SetUp()
        {
            var tableDef = new TableDefinition("Test");
            var tableBuilder = new TableDefiner(tableDef);
            tableBuilder.CompareKey("Key");
            _table = new SnapshotTable(tableDef);
        }

        [Test]
        public void RowsCanBeAdded()
        {
            //Act
            var row = _table.AddRow();

            //Assert
            row.Should().BeOfType<SnapshotRow>();
        }

        [Test]
        public void RowsAreReported()
        {
            //Arrange
            var output = new Output();
            for (var i = 1; i <= 10; i++)
            {
                var row = _table.AddRow();
                row.SetField("Key", i);
                row.SetField("Surname", $"Surname {i}");
                row.SetField("FirstName", $"First Name {i}");
                row.SetField("TimeStamp", DateTime.Parse($"2020-03-21 11:06:{i:00}"));
                row.SetField("Double", (double)i / 3);
                row.SetField("Decimal", Math.Round((decimal)i / 3, 2));
                row.SetField("Bytes", Enumerable.Range(1, i).Select(n => (byte)i).ToArray());
            }

            //Act
            _table.ReportContents(output);
            
            //Assert
            output.Report.Verify();

        }
    }
}
