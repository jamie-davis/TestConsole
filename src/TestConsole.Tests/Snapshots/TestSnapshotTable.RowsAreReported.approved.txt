Test                                                                                           
----                                                                                           
     Key Surname      FirstName     TimeStamp                 Double      Decimal Bytes        
-------- ------------ ------------- ------------------- ------------ ------------ -------------
       1 Surname 1    First Name 1  21/03/2020 11:06:01         0.33         0.33 System.Byte[]
       2 Surname 2    First Name 2  21/03/2020 11:06:02         0.67         0.67 System.Byte[]
       3 Surname 3    First Name 3  21/03/2020 11:06:03         1.00         1.00 System.Byte[]
       4 Surname 4    First Name 4  21/03/2020 11:06:04         1.33         1.33 System.Byte[]
       5 Surname 5    First Name 5  21/03/2020 11:06:05         1.67         1.67 System.Byte[]
       6 Surname 6    First Name 6  21/03/2020 11:06:06         2.00         2.00 System.Byte[]
       7 Surname 7    First Name 7  21/03/2020 11:06:07         2.33         2.33 System.Byte[]
       8 Surname 8    First Name 8  21/03/2020 11:06:08         2.67         2.67 System.Byte[]
       9 Surname 9    First Name 9  21/03/2020 11:06:09         3.00         3.00 System.Byte[]
      10 Surname 10   First Name 10 21/03/2020 11:06:10         3.33         3.33 System.Byte[]
                                                                                               